app/controllers/bookmarks/search_controller.rb

```diff
@@ -1,19 +1,85 @@
 class Bookmarks::SearchController < ApplicationController
+  require_login!
+  before_action :set_search, only: [ :show, :edit, :update, :destroy ]
+
   # GET /bookmarks/search
   # GET /bookmarks/search.json
   def index
-    @bookmarks = BookmarksIndex.query(
-      bool: {
-        should: [
-          { match: { title: params[:q] } },
-          { match: { tags: params[:q] } }
-        ]
-      }
-    ).objects.page params[:page]
+    @bookmarks = fetch_bookmarks default_query
 
     respond_to do |format|
       format.html { render :index }
       format.json { render :index, status: :ok }
     end
   end
+
+  # POST /bookmarks/search
+  # POST /bookmarks/search.json
+  # rubocop:disable Metrics/AbcSize
+  def create
+    @query = params[:query].to_unsafe_h
+
+    @search = authorize current_user.searches.new(query: @query)
+
+    respond_to do |format|
+      if @search.save
+        format.html { redirect_to bookmarks_search_path(@search) }
+        format.json { render :show, status: :created, location: bookmarks_search_path(@search) }
+      else
+        format.html { render :new }
+        format.json { render json: @search.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+  # rubocop:enable Metrics/AbcSize
+
+  # GET /bookmarks/search/1
+  # GET /bookmarks/search/1.json
+  def show
+    @query = @search.query.with_indifferent_access
+    @bookmarks = fetch_bookmarks(@query)
+
+    respond_to do |format|
+      format.html { render :index }
+      format.json { render :index, status: :ok }
+    end
+  end
+
+  protected
+
+  def set_search
+    @search = authorize current_user.searches.find(params[:id])
+  end
+
+  # rubocop:disable Naming/MemoizedInstanceVariableName
+  def default_query
+    return @query ||= {} if params[:q].blank? || params[:q].empty?
```

Comment:
this logic is weird

Reply:
This feature is weird. Its a fallback for when no js is enabled but I should see about encapsulating it into the searcher class, maybe?